source/frontend/src/components/common/HeaderField/StatusField.tsx

```diff
@@ -0,0 +1,138 @@
+import { Col, Row } from 'react-bootstrap';
+import { IconType } from 'react-icons';
+import { BiDuplicate } from 'react-icons/bi';
+import { FaCheck, FaPauseCircle, FaStopCircle } from 'react-icons/fa';
+import { MdArchive, MdCancel, MdCircle, MdEdit } from 'react-icons/md';
+import styled from 'styled-components';
+
+import { Dictionary } from '@/interfaces/Dictionary';
+import { ApiGen_Base_CodeType } from '@/models/api/generated/ApiGen_Base_CodeType';
+
+import { InlineFlexDiv } from '../styles';
+
+interface IStatusFieldProps {
+  statusCodeType: ApiGen_Base_CodeType<string>;
+}
+
+interface StatusStyle {
+  icon: IconType;
+  colorVariant: string;
+}
+
+const statusDictionary: Dictionary<StatusStyle> = {
+  active: { icon: MdCircle, colorVariant: 'green' },
+  archived: { icon: MdArchive, colorVariant: 'grey' },
+  completed: { icon: FaCheck, colorVariant: 'blue' },
+  hold: { icon: FaPauseCircle, colorVariant: 'yellow' },
+  cancelled: { icon: MdCancel, colorVariant: 'red' },
+  draft: { icon: MdEdit, colorVariant: 'yellow' },
+  terminated: { icon: FaStopCircle, colorVariant: 'red' },
+  duplicate: { icon: BiDuplicate, colorVariant: 'blue' },
+};
+
+const StatusField: React.FunctionComponent<React.PropsWithChildren<IStatusFieldProps>> = ({
+  statusCodeType,
+}) => {
+  const translateStatusCode = (statusCodeType: ApiGen_Base_CodeType<string>) => {
+    switch (statusCodeType.id.toLowerCase()) {
+      case 'active':
+      case 'ac':
+        return statusDictionary['active'];
+      case 'archived':
+      case 'cncn':
+      case 'archiv':
+        return statusDictionary['archived'];
+      case 'cancelled':
+      case 'ca':
+      case 'cancel':
+      case 'discard':
+        return statusDictionary['cancelled'];
+      case 'complete':
+      case 'closed':
+      case 'co':
+      case 'complt':
+        return statusDictionary['completed'];
+      case 'draft':
+      case 'pl':
+        return statusDictionary['draft'];
+      case 'duplicate':
+        return statusDictionary['duplicate'];
+      case 'hold':
+      case 'inactive':
+      case 'ho':
+        return statusDictionary['hold'];
+      case 'terminated':
+        return statusDictionary['terminated'];
+      default:
+        break;
+    }
+    return statusDictionary['cancelled'];
+  };
+
+  const statusFound = translateStatusCode(statusCodeType);
+  return (
+    <StyledBottomRow className="no-gutters justify-content-end align-items-end">
+      <Col />
+      <Col xs="auto" className="align-self-end">
+        <RetiredWarning $variant={statusFound.colorVariant}>
```

Comment:
a prop with a name starting with a dollar sign seems new, what does that naming indicate?

Reply:
I set it like to differentiate it from a react component (that one is a style). I dont feel strongly about it so will change if it helps